### PR TITLE
Moved 180 Ringing response from core to app (menu module)

### PR DESCRIPTION
**Pull request by juha-h**
Required by https://github.com/baresip/baresip/pull/3088

### CODE REVIEW

File: src/sipsess/accept.c [hunk 1]
```diff
@@ -74,8 +74,8 @@ int sipsess_accept(struct sipsess **sessp, struct sipsess_sock *sock,
 	va_list ap;
 	int err;
 
-	if (!sessp || !sock || !msg || scode < 101 || scode > 299 ||
-	    !cuser || !ctype)
+	if (!sessp || !sock || !msg || (scode != 0 && scode < 100) ||
+	    scode > 299 || !cuser || !ctype)
```
Comment (cspiel1) [inline src/sipsess/accept.c:78]: I am not sure with this. Why not directly use `sipsess_alloc()`?

Comment (juha-h) [reply src/sipsess/accept.c:78]: I'll  investigate.

Comment (juha-h) [reply src/sipsess/accept.c:78]: `sipsess_alloc()` is not in re API.

Comment (juha-h) [reply src/sipsess/accept.c:78]: @cspiel1 What is wrong with `sipsess_accept()`? In addition to `sipsess_alloc()` it does lots other stuff that would also need to be handled separately:
```
err = sip_dialog_accept(&sess->dlg, msg);
	if (err)
		goto out;

	hash_append(sock->ht_sess,
		    hash_joaat_str(sip_dialog_callid(sess->dlg)),
		    &sess->he, sess);

	sess->msg = mem_ref((void *)msg);

	err = sip_strans_alloc(&sess->st, sess->sip, msg, cancel_handler,
			       sess);
```
So I would prefer to keep the PR as it currently is.  I have made tests and they all have worked fine.

Comment (cspiel1) [reply src/sipsess/accept.c:78]: In my opinion `sipsess_accept()` should send a reply. @alfredh and @sreimers please review and decide!

Should we revert this, when we have another solution?

**Comment by juha-h** [comment]
Christian Spielberger writes:

> In my opinion `sipsess_accept()` should send a reply. @alfredh and @sreimers please review and decide!

If so, we can sent 100 Trying like it was.

> Should we revert this, when we have another solution?

It is OK to me to revert this IF the other solution fixes the bug, i.e.,
does not by default send 180 Ringing response.

**Comment by maximilianfridrich** [comment]
The 100 Trying response was not accepted before. See response from Alfred:

> please read the spec carefully regarding 100 usage and 101-199 replies:
> https://datatracker.ietf.org/doc/html/rfc3261#section-21.1.1
> 100 Trying should only be sent by a SIP Proxy...

https://github.com/baresip/re/pull/808#issuecomment-1550970325

**Comment by juha-h** [comment]
Maximilian Fridrich writes:

> The 100 Trying response was not accepted before. See response of Alfred:
> 
> > please read the spec carefully regarding 100 usage and 101-199 replies:
> > https://datatracker.ietf.org/doc/html/rfc3261#section-21.1.1
> > 100 Trying should only be sent by a SIP Proxy...
> 
> https://github.com/baresip/re/pull/808#issuecomment-1550970325

RFC text uses term "next hop server" that very well can be an UAS = User
Agent Server.  See for example this Cisco example call flow:

https://community.cisco.com/t5/collaboration-knowledge-base/basic-sip-call-flows-amp-troubleshooting-commands/ta-p/3110162

But this PR does not currently send a 100 Trying response.

Please commit this PR or provide another one that does not by default
send 180 Ringing response to INVITE request.

**Comment by cspiel1** [comment]
@juha-h Can we close this?

**Comment by juha-h** [comment]
Yes, forgot about the re PR when I closed the baresip PR.

**Pull request closed without merging**